### PR TITLE
delphi client template

### LINKED ISSUE #35: Implement Oracle receiving and sending messages to Delphi
Make request to delphi to get node/graph and position information

### PR DESCRIPTION
**Pull request by svnphlps**
close #35

### CODE REVIEW

File: oracle/Assets/Scripts/DelphiClient.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+using SimpleJSON;
+
+public class DelphiClient : MonoBehaviour
+{
+  //   todo: remove
+  //   private readonly string EX_GET_REQUEST_URL = "https://dog.ceo/api/breed/akita/images/random";
+  //   private readonly string EX_POST_REQUEST_URL = "https://reqres.in/api/users";
+
+  // todo: update w/ actual url
+  private readonly string GET_NODES_URL = "";
```
Comment (pgayee694) [inline oracle/Assets/Scripts/DelphiClient.cs]: Is there a way we can pass command line arguments to the Unity app to give it the url of delphi? @mooncollin

Comment (pgayee694) [reply oracle/Assets/Scripts/DelphiClient.cs]: using a config file

File: oracle/Assets/Scripts/DelphiClient.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  public void UpdateStatus(int id, bool status)
+  {
+    StartCoroutine(CallUpdateStatus(id, status));
+  }
+
+  //client logic
+  IEnumerator CallGetStatuses(int[] roomIds)
+  {
+    UnityWebRequest www = UnityWebRequest.Get(BASE_URL + "/get-statuses?room_ids=" + string.Join(",", roomIds));
```
Comment (pgayee694) [inline oracle/Assets/Scripts/DelphiClient.cs]: the url needs '&' between each room id
i.e.  "/get-statuses?room_id=260&room_id=261"

File: oracle/Assets/Scripts/DelphiClient.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    JSONNode response = JSON.Parse(www.downloadHandler.text);
+    print(response);
+
+    // todo: push status into local object
+    // ex: local256.status = response[256]
+  }
+
+  IEnumerator CallGetCounts(int[] roomIds)
+  {
+    UnityWebRequest www = UnityWebRequest.Get(BASE_URL + "/get-counts?room_ids=" + string.Join(",", roomIds));
```
Comment (pgayee694) [inline oracle/Assets/Scripts/DelphiClient.cs]: same as above

**Review by pgayee694 — CHANGES_REQUESTED** [review]

**Review by mooncollin — APPROVED** [review]

**Comment by pgayee694** [comment]
looks good, just fix your rebase or merge master into your branch

**Comment by pgayee694** [comment]
provide a default config.ini

**Review by pgayee694 — APPROVED** [review]

**Pull request merged**